flink-runtime/src/test/java/org/apache/flink/runtime/memory/MemoryManagerTest.java

```diff
@@ -298,6 +299,39 @@ public void testMemoryAllocationAndReservation() throws MemoryAllocationExceptio
 		memoryManager.releaseAllMemory(owner2, type);
 	}
 
+	@Test
+	public void testComputeMemorySize() {
+		double fraction = 0.6;
+		assertEquals((long) (memoryManager.getMemorySize() * fraction), memoryManager.computeMemorySize(fraction));
+
+		fraction = 1.0;
+		assertEquals((long) (memoryManager.getMemorySize() * fraction), memoryManager.computeMemorySize(fraction));
+	}
+
+	@Test
+	public void testComputeMemorySizeInvalidFraction() {
+		try {
+			memoryManager.computeMemorySize(0.0);
```

Comment:
Could we do like this instead for each case?
```
@Test(Expected = IllegalArgumentException.class)
public void testComputeMemorySizeFailForZeroFraction() {
    memoryManager.computeMemorySize(0.0);
}
```

Reply:
Sure.

Reply:
done.